.ci/README.md

```diff
@@ -38,20 +38,49 @@ or
+### Divergence by zero commits
+
+Just click retry on the failed job in Cloud Build.  Yay!
+
+### Divergence by exactly one commit.
+
+Find which commit caused the error.  This will usually be easy - cloud build lists the commit which triggered a build, so you can probably just use that one.  You need to set all the sync tags to the parent of that commit.  Say the commit which caused the error is `12345abc`.  You can find the parent of that commit with `git rev-parse 12345abc~` (note the `~` suffix).  Some of the sync tags are likely set to this value already.  For the remainder, simply perform a git push.  Assuming that the parent commit is `98765fed`, that would be, e.g. `git push origin 98765fed:tf-conv-sync`.
 
 If you are unlucky, there may be open PRs - this only happens if the failure occurred during the ~5 second period surrounding the merging of one of the downstreams.  Close those PRs before proceeding to the final step.
 
-Click "retry" on the failed job in Cloud Build.  Watch the retried job and see if it succeeds - it should!  If it does not, the underlying problem may not have been fixed.
+Finally, click "retry" on the failed job in Cloud Build.  Watch the retried job and see if it succeeds - it should!  If it does not, the underlying problem may not have been fixed.
+
+### Divergence by more than one commit.
+This situation is interesting.  This means that there was a failure but that it went undetected for a little while, or that there was a failure in a PR which was merged at about the same time as another one.  Don't worry, it's still easy to fix!  To understand this, you should know a little more about the way the pusher tasks work.  For each commit merged in Magic Modules, five pusher tasks (part of one pusher job) start.  Each pusher task waits for all previous pusher tasks of the same type to finish.  That is, the TPGB pusher always waits for all previous TPGB pushers, the Ansible pusher waits for all previous Ansible pushers, etc.
+
+When a task fails, all the other tasks which are part of the same job are cancelled.  However, if one of those tasks has already succeeded, the next task of the same kind - the next Ansible pusher, say, since Ansible tends to finish quickly - will be free to start.  That means that all subsequent jobs have partially succeeded - they pushed Ansible, but not TPGB.  You'll need to do the work of the failed tasks yourself in order to restore the steady state.
+
+When this happened the first time, Cameron and Nathan wrote this little shell snippet, which should do it for you.  You will need to get the Magician's github token, either by generating a new one (be sure to clean up after yourself when done) or by decrypting the value in .ci/gcb-push-downstream.yml.
```

Comment:
Is the token also available via valentine? Or was that something else

Reply:
No, unfortunately.  I could add my new one, though...

Reply:
I did add it, so let me update this to say you can fetch it from there...